lib/utils/report.cjs

```diff
@@ -162,41 +173,144 @@ function isDisabled(ruleName, startLine, disabledRanges) {
 	return false;
 }
 
-/** @param {Problem & { line: number }} problem */
-function isFixApplied({ fix, line, result: { stylelint }, ruleName }) {
-	const { disabledRanges, config = {}, fixersData } = stylelint;
+/**
+ * @param {Problem & { line: number }} problem
+ * @returns {{range: PostcssRange, text: string, applied: false} | {applied: boolean}}
+ */
+function applyFix({ fix, fixedNode, line, result: { stylelint }, ruleName }) {
+	const { disabledRanges, config = {}, fixersData, fixedRanges } = stylelint;
+
+	if (!validateTypes.isFunction(fix)) return { applied: false };
 
-	if (!validateTypes.isFunction(fix)) {
-		addFixData({ fixersData, ruleName, fixed: false });
+	const shouldFix = Boolean(config.fix || (config.computeReplacementText && fixedNode));
+	const shouldFixThisRule = Boolean(shouldFix && !config.rules?.[ruleName][1]?.disableFix);
+	const mayFix =
+		shouldFixThisRule && (config.ignoreDisables || !isDisabled(ruleName, line, disabledRanges));
+
+	if (!mayFix) return { applied: false };
+
+	fix();
+
+	if (!config.computeReplacementText) {
+		addFixData({ fixersData, ruleName });
 
-		return false;
+		return { applied: true };
 	}
 
-	const shouldFix = Boolean(config.fix && !config.rules?.[ruleName][1]?.disableFix);
-	const mayFix =
-		shouldFix && (config.ignoreDisables || !isDisabled(ruleName, line, disabledRanges));
+	if (!fixedNode?.source?.start || !fixedNode?.source?.end) {
+		return { applied: false };
+	}
 
-	addFixData({ fixersData, ruleName, fixed: mayFix });
+	const fixedNodeRange = {
+		start: fixedNode.source.start,
+		end: fixedNode.source.end,
+	};
 
-	if (!mayFix) return false;
+	// When recording replacement text we want to ensure that there is no overlap with any other fix.
+	// We only record the first fix for each node.
+	if (fixedRanges.some((fixedRange) => rangesOverlap(fixedRange, fixedNodeRange)))
+		return { applied: false };
```

Comment:
> We only record the first fix for each node.

Looking at https://github.com/stylelint/stylelint/pull/8234/files#diff-b429bc70b09564c65d1a7f4436fa67271400e65e9fbaf87262a9217f966eeee4R304 the description may be inaccurate.
i.e. it's not necessarily the same node (i.e. it may be a child or even a sibling if the previous fix was significant)

---

More importantly the  [`"text"`](https://github.com/stylelint/stylelint/pull/8234/files#diff-9a4b261e295bef1ccce813aa194ac4023a1dcf98236667889964b1ba6fbc551fR176) of the already stored `fixedData` has the text that corresponds to that previous state at the time of that "first" fix which takes precedence.

Hence if a latter fix falls within that range the `text` will not reflect the state after the following fixes.
i.e. before the `return` at l212 we have adjust the `text` even if we keep the `range` untouched
If you diff after the lint has been done all the matching `fixData.range` must be updated.
If you diff per report then only the following ones must have need `fixData`.

Of course all of this matters only if you are planning to store the `text` alongside its corresponding `range` for a later retrieval;
 which seems to be the case looking at https://github.com/stylelint/stylelint/pull/8234/files#diff-cba25f954347e963307879ca83b95b319e9003d9b4b51fd350d3f4110566d3a8R120

tl;dr:

I think that the condition `if ('range' in fixData)` of
```
	if ('range' in fixData) {
		warningProperties.fix = {
			range: fixData.range,
			text: fixData.text,
		};
	}
```
prevents us from updating adding `warningProperties.fix` for the overlap cases because we failed to return a new `fixData` with the updated `text`.
i.e. the non-"first" get no `warningProperties.fix` and even if they were it would be inaccurate if not modified

Did I get something wrong?
Am I overlooking something?

Reply:
> i.e. it's the `text` before any fixes


`Warning.fix.text` is the replacement text, not the original text.

- `replacement text` is any text generated by a fixer, intended to be inserted by another tool
- `original text` is the text provided to Stylelint, before any mutations